tachyon/math/finite_fields/goldilocks/internal/goldilocks_prime_field_x86_special.cc

```diff
@@ -37,6 +37,16 @@ CLASS CLASS::MinusOne() {
 #endif
 }
 
+// static
+template <typename Config>
+CLASS CLASS::TwoInv() {
+#if USE_MONTGOMERY == 1
+  return PrimeField::FromMontgomery(Config::kTwoInv[0]);
+#else
+  return PrimeField(Config::kTwoInv[0]);
```

Comment:
Since `Config::kTwoInv` is a unreduced montgomery form, so you should compute this directly!

```suggestion
  return unwrap(PrimeField(2).Inverse());
```